editor/editor_properties_array_dict.cpp

```diff
@@ -429,6 +429,12 @@ void EditorPropertyArray::_button_draw() {
 bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 	String allowed_type = Variant::get_type_name(subtype);
 
+	// When the subtype is of type Object, an additional subtype may be specified in the hint string
+	// (e.g. Resource, Texture2D, ShaderMaterial, etc). We want the allowed type to be that, not just "Object".
+	if (subtype == Variant::OBJECT && subtype_hint_string != "") {
+		allowed_type = subtype_hint_string;
```

Comment:
Are you sure that `subtype_hint_string` will always be a valid object type?

What about these for example:
```
scene/main/canvas_item.cpp
959:    ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "material", PROPERTY_HINT_RESOURCE_TYPE, "CanvasItemMaterial,ShaderMaterial"), "set_material", "get_material");

scene/resources/environment.cpp
1358:   ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "adjustment_color_correction", PROPERTY_HINT_RESOURCE_TYPE, "Texture2D,Texture3D"), "set_adjustment_color_correction", "get_adjustment_color_correction");
```

Or possible use of the weird `17/17:RichTextEffect` syntax (currently seen on `Variant::ARRAY` but who knows..).

Reply:
Yeah so this is for the array editor, those appear to just be normal properties. The "Subtype" is the first number before the `/` I believe, and at the moment that is only ever for `Variant::OBJECT.` (currently `21`)

Reply:
Like discussed before this should really be changed, but for now this resolves it, lol.